cmds/rush/tty_linux.go

```diff
@@ -52,9 +53,9 @@ func tty() {
 func foreground() {
 	// Place process group in foreground.
 	if ttypgrp != 0 {
-		r1, r2, errno := syscall.RawSyscall(syscall.SYS_IOCTL, ttyf.Fd(), uintptr(syscall.TIOCSPGRP), uintptr(unsafe.Pointer(&ttypgrp)))
+		_, _, errno := unix.RawSyscall(unix.SYS_IOCTL, ttyf.Fd(), uintptr(unix.TIOCSPGRP), uintptr(unsafe.Pointer(&ttypgrp)))
```

Comment:
What do you mean by "doesn't allow"? Is it not possible to pass int(unsafe.Pointer(&ttypgrp)) to unix.IoctlSetInt?

Reply:
Sorry for not being more specific. By "doesn't allow" I meant there's no dedicated function to set a value from pointer. I guess `int(uintptr(unsafe.Pointer(&ttypgrp)))` should work (unsafe.Pointer can only be converted to uintptr), but I'm not sure whether the conversion from 64 bit to int might pose problems because int is not defined in size, only that it is at least 32 bits.

I guess the safest option would be to add something like `IoctlSetUintptr(fd int, req uint, ptr uintptr)` to x/sys/unix which we then can use.

What do you think?

Reply:
You're right, I think there should be an `IoctlSetUintptr`. `int` would probably work fine, since it's 64bit on most 64bit systems, but with `uintptr` comes the guarantee.